cpp/ql/lib/semmle/code/cpp/commons/Buffer.qll

```diff
@@ -33,18 +33,18 @@ predicate memberMayBeVarSize(Class c, MemberVariable v) {
   ) and
   // If the size is taken, then arithmetic is performed on the result at least once
   (
+    // `sizeof(c)` is not taken
     not exists(SizeofOperator so |
-      // `sizeof(c)` is taken
       so.(SizeofTypeOperator).getTypeOperand().getUnspecifiedType() = c or
       so.(SizeofExprOperator).getExprOperand().getUnspecifiedType() = c
     )
     or
+    // or `sizeof(c)` is taken
     exists(SizeofOperator so |
-      // `sizeof(c)` is taken
       so.(SizeofTypeOperator).getTypeOperand().getUnspecifiedType() = c or
       so.(SizeofExprOperator).getExprOperand().getUnspecifiedType() = c
     |
-      // arithmetic is performed on the result
+      // and arithmetic is performed on the result
       so.getParent*() instanceof AddExpr
```

Comment:
I imagine this `AddExpr` should really be a `PointerArithmeticOperation` (to also include stuff like _subtracting_ a constant from the pointer, or computing a difference between two pointers). But obviously, this PR is not the place to do this.

Reply:
This expression is to detect allocations on the form:
`struct A *a = malloc(sizeof(struct A) + 100)` 

But maybe we want to mark arrays of size <= 1 as variable sized is the said field is used in conjunction with pointer arithmetic.

Reply:
Oh, you're right! I misread that. Thanks for clearing up my confusion. And yes, I agree that we could use that as another heuristic for detecting more variable sized arrays.